sdk/selfhelp/Azure.ResourceManager.SelfHelp/src/Generated/TroubleshooterResourceData.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Collections.Generic;
+using Azure.Core;
+using Azure.ResourceManager.Models;
+using Azure.ResourceManager.SelfHelp.Models;
+
+namespace Azure.ResourceManager.SelfHelp
+{
+    /// <summary>
+    /// A class representing the TroubleshooterResource data model.
+    /// Troubleshooter response.
+    /// </summary>
+    public partial class TroubleshooterResourceData : ResourceData
+    {
+        /// <summary> Initializes a new instance of TroubleshooterResourceData. </summary>
+        public TroubleshooterResourceData()
+        {
+            Parameters = new ChangeTrackingDictionary<string, string>();
+            Steps = new ChangeTrackingList<Step>();
+        }
+
+        /// <summary> Initializes a new instance of TroubleshooterResourceData. </summary>
+        /// <param name="id"> The id. </param>
+        /// <param name="name"> The name. </param>
+        /// <param name="resourceType"> The resourceType. </param>
+        /// <param name="systemData"> The systemData. </param>
+        /// <param name="solutionId"> Solution Id to identify single troubleshooter. </param>
+        /// <param name="parameters"> Client input parameters to run Troubleshooter Resource. </param>
+        /// <param name="provisioningState"> Status of troubleshooter provisioning. </param>
+        /// <param name="steps"> List of step object. </param>
+        internal TroubleshooterResourceData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, string solutionId, IDictionary<string, string> parameters, TroubleshooterProvisioningState? provisioningState, IReadOnlyList<Step> steps) : base(id, name, resourceType, systemData)
+        {
+            SolutionId = solutionId;
+            Parameters = parameters;
+            ProvisioningState = provisioningState;
+            Steps = steps;
+        }
+
+        /// <summary> Solution Id to identify single troubleshooter. </summary>
+        public string SolutionId { get; set; }
```

Comment:
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).

Reply:
@archerzz  solutionId is neither GUId/ResourceType but string